packages/core/src/utils.ts

```diff
@@ -17,15 +17,33 @@ export const encodeQueryParameters = (baseUrl: string, parameters?: Record<strin
   else return baseUrl
 }
 
-export const rewriteRedirectTo = <T extends RedirectOption>(clientUrl: string, options?: T) =>
-  options?.redirectTo
-    ? {
-        ...options,
-        redirectTo: options?.redirectTo?.startsWith('/')
-          ? clientUrl + options.redirectTo
-          : options?.redirectTo
-      }
-    : options
+export const rewriteRedirectTo = <T extends RedirectOption>(clientUrl: string, options?: T) => {
+  if (options?.redirectTo) {
```

Comment:
I don't know if it's just me, but reversing the logic of if statements here makes it easier on the eye to me:

```ts
if (!options?.redirectTo) {
  return options
}

// then we can proceed without the else branch...
const baseClientUrl = new URL(clientUrl)
// ...
```